src/promoCode.d.ts

```diff
@@ -19,11 +19,11 @@ export type HubbyPromoCode = {
   valid_to: string | Date | Timestamp;
   booking: DocumentReference<Booking> | null;
   external_id: string;
-  type?: string;
-  countries?: string[]; 
-  max_bytes?: number; 
-  usage?: string[], 
-  uuid_usage?: string[], 
+  type?: 'full-discount' | 'partial-discount' | 'booking' | 'traveler' | null | string;
```

Comment:
i see we have a no-cost type in the code that i've been refactoring. I assume those are covered by booking and traveler, but there is also a referral, do we need that one here as well?

Reply:
Added referral